system/Database/Query.php

```diff
@@ -275,35 +275,24 @@ protected function compileBinds()
     {
         $sql = $this->finalQueryString;
 
-        $hasBinds      = strpos($sql, $this->bindMarker) !== false;
-        $hasNamedBinds = ! $hasBinds
-            && preg_match('/:(?!=).+:/', $sql) === 1;
-
-        if (empty($this->binds)
-            || empty($this->bindMarker)
-            || (! $hasNamedBinds && ! $hasBinds)
-        ) {
+        if (empty($this->binds)) {
             return;
         }
 
-        if (! is_array($this->binds)) {
-            $binds     = [$this->binds];
-            $bindCount = 1;
-        } else {
-            $binds     = $this->binds;
+        $binds = is_array($this->binds) ? $this->binds : [$this->binds];
```

Comment:
```suggestion
        $binds = (array) $this->binds;
```

Reply:
I can't think of any regressions with that change, so why not.